components/remote-environment-broker/internal/syncer/ns_broker_syncer_test.go

```diff
@@ -0,0 +1,122 @@
+package syncer
+
+import (
+	"testing"
+
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/fake"
+	"github.com/stretchr/testify/assert"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	k8s_testing "k8s.io/client-go/testing"
+)
+
+func TestServiceBrokerSync_Success(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset(fixServiceBroker())
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Nil(t, err)
+}
+
+func TestServiceBrokerSync_Empty(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset()
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Nil(t, err)
+}
+
+func TestServiceBrokerSync_ListError(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset()
+	client.PrependReactor("list", "servicebrokers", failingReactor(fixError()))
+
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Error(t, err)
+}
+
+func TestServiceBrokerSync_ConflictError(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset(fixServiceBroker())
+	client.PrependReactor("update", "servicebrokers", failingReactor(apiErrors.NewConflict(schema.GroupResource{}, "", fixError())))
+
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Error(t, err)
+}
+
+func TestServiceBrokerSync_UpdateError(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset(fixServiceBroker())
+	client.PrependReactor("update", "servicebrokers", failingReactor(fixError()))
+
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Error(t, err)
+}
+
+func TestServiceBrokerSync_GetError(t *testing.T) {
+	// given
+	client := fake.NewSimpleClientset(fixServiceBroker())
+	client.PrependReactor("get", "servicebrokers", failingReactor(fixError()))
+
+	nsBrokerSync := NewServiceBrokerSyncer(client.Servicecatalog())
+
+	// when
+	err := nsBrokerSync.Sync(fixLabelSelector(), maxSyncRetries)
+
+	// then
+	assert.Error(t, err)
```

Comment:
be more specific when making assertions against errors

Reply:
it's a muliterror as discussed.